eZ/Publish/Core/Repository/SectionService.php

```diff
@@ -203,25 +204,19 @@ public function loadSection($sectionId)
     }
 
     /**
-     * Loads all sections.
-     *
-     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the current user user is not allowed to read a section
+     * Loads all sections, excluding the ones the current user is not allowed to read.
      *
      * @return \eZ\Publish\API\Repository\Values\Content\Section[]
      */
     public function loadSections()
     {
-        $sections = [];
-        foreach ($this->sectionHandler->loadAll() as $spiSection) {
-            $sections[] = $section = $this->buildDomainSectionObject($spiSection);
-
-            // @todo change API to just filter instead of throwing here
-            if (!$this->permissionResolver->canUser('section', 'view', $section)) {
-                throw new UnauthorizedException('section', 'view');
-            }
-        }
+        $sections = array_map(function ($spiSection) {
```

Comment:
`array_map([$this, 'buildDomainSectionObject'], $this->sectionHandler->loadAll());`